metron-deployment/packaging/ambari/metron-mpack/README.md

```diff
@@ -0,0 +1,127 @@
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+This provides a Management Pack (MPack) extension for [Apache Ambari](https://ambari.apache.org/) that simplifies the provisioning, management and monitoring of Metron on clusters of any size.  
+
+This allows you to easily install Metron using a simple, guided process.  This also allows you to monitor cluster health and even secure your cluster with kerberos.
+
+### Prerequisites
+
+* Ambari 2.4.2+
+
+* Installable Metron packages (either RPMs or DEBs) located in a repository on each host at `/localrepo`.
+
+* A [Node.js](https://nodejs.org/en/download/package-manager/) repository installed on the host running the Management and Alarm UI.
+
+### Quick Start
+
+1. Build the Metron MPack. Execute the following command from the project's root directory.
+    ```
+    mvn clean package -Pmpack -DskipTests
+    ```
+
+1. This results in the Mpack being produced at the following location.
+    ```
+    metron-deployment/packaging/ambari/metron-mpack/target/metron_mpack-x.y.z.0.tar.gz
+    ```
+
+1. Copy the tarball to the host where Ambari Server is installed.
+
+1. Ensure that Ambari Server is stopped.
+
+1. Install the MPack.
+    ```
+    ambari-server install-mpack --mpack=metron_mpack-x.y.z.0.tar.gz --verbose
+    ```
+
+1. Install the Metron packages (RPMs or DEBs) in a local repository on each host where a Metron component is installed.  By default, the repository is expected to exist at `/localrepo`.
+
+    On hosts where only a Metron client is installed, the local repository must exist, but it does not need to contain Metron packages.  For example to create an empty repository for an RPM-based system, run the following commands.
+
+    ```
+    yum install createrepo
+    mkdir /localrepo
+    cd /localrepo
+    createrepo
+    ```
+
+1. Metron will now be available as an installable service within Ambari.  
+
+### Installation Notes
+
+The MPack will make all Metron services available in Ambari in the same manner as any other services in a stack.  These can be installed using Ambari's user interface using "Add Services" or during an initial cluster install.
+
+#### Co-Location
+
+1. The Parsers, Enrichment, Indexing, and Profiler masters should be colocated on a host with a Kafka Broker.  This is necessary so that the correct Kafka topics can be created.
+
+1. The Enrichment and Profiler masters should be colocated on a host with an HBase client.  This is necessary so that the Enrichment, Threat Intel, and Profile tables can be created.
+
```

Comment:
Here are a couple more co-location requirements:
3. Enrichment, Indexing, Profiler should be co-located on the same host as Parsers.
4. Alerts UI and Management UI should be co-located on the same host.
5. Parsers and REST need to have an instance of STORM SUPERVISOR client

Reply:
Just to give some more context on the co-location requirements...

The point no. 3 above is due to the way the [service_advisor](https://github.com/apache/metron/blob/master/metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/service_advisor.py#L78) is designed. This checks for all of the Parsers, Enrichment, Indexing and Profiler to be on the same host.